cheroot/workers/threadpool.py

```diff
@@ -180,8 +183,15 @@ def put(self, obj):
                 waiting to be processed
         """
         self._queue.put(obj, block=True, timeout=self._queue_put_timeout)
-        if obj is _SHUTDOWNREQUEST:
-            return
+
+    def _clear_dead_threads(self):
+        # Remove any dead threads from our list
+        for t in [t for t in self._threads if not t.is_alive()]:
+            self._threads.remove(t)
+            try:
```

Comment:
This is a perfect case for contextlib usage
```suggestion
            with contextlib.suppress(IndexError):
```

Reply:
Fine with me.